department-of-veterans-affairs/caseflow
# Kamalam7/appeals 13369 v1

Resolves APPEALS-13369

Description
Create Send initial notification letter from Post Send Initial Notification Letter Holding Task. In the Clerk of the board view the Post Send Initial Notification Letter Holding Task appears on the "on hold" tab. If we click on the Case details, in the Currently active task we can see the Actions where we can select the drop down from the actions and click on Resend initial notification letter. A modal pop up appears for Resend initial notification letter and when we click on Resend initial letter button, the case routed to "unassigned" tab in the Clerk of the board view. In the unassigned tab the task has to change as Send Initial Notification Letter Task. And when we click on case details, in the Currently active task, Task will be shown as Send Initial Notification Letter Task. If we go to Case timeline it marks Post Send Initial Notification Letter Holding Task is completed and shows the hold time period. 

Acceptance Criteria

[x] Code compiles correctly

[x] The following occurs when the “Resend Initial Notification Letter” task action is selected (used when the Send Initial Notification Letter needs to be reprocessed)
Modal appears asking user to confirm that they want to resend initial notification letter
User is required to provide a reason why the letter needs to be resent
Field type: free text with a 350-character limit

[x] If confirmed, The “Post-Send Initial Notification Letter Holding Period” task is marked as complete
      Task moves from Currently Active Tasks section to Case Timeline and contains the following 
             Date of occurrence
             Assigned to: <organization>
             Assigned by: <user who completed action to assign task to COB>
             Task Instructions
             period of time in which task was on hold (e.g., 23/45 days)
             <user input in the Resend Initial Notification Letter modal>

[x] A new “Send Initial Notification Letter” task from  is created and active
     Task appears in the Currently Active Tasks section
     Task is assigned to the Clerk of the Board organization

[x] Case appears in the Unassigned tab of the Clerk of the Board organizational queue


Testing Plan

1. Go to Caseflow Login and Sign as BVADWISE
2. Navigate to Mail Intake 
3. From the dropdown select VA FORM 10182 for processing and click on continue to search
4. Enter a veteran file number. I used 711790008. Click the search icon to continue
5. Fill out the form and select direct review for the review option type. Click the button at the bottom to continue
6. On the add issues page, click the add issues button. Select a benefit type as Compensation and Issue Category as "Contested claims - Attorney Fees. Click on Add this issue and establish appeal to complete the intake.
7. The intake should be completed. 
8. Switch user and select Clerk of the board user and enter into Queue and click on Switch view and select Clerk of the board team view
<img width="881" alt="image" src="https://user-images.githubusercontent.com/110078646/221021481-6a22df82-4bee-4bff-89fd-68d00dc2da39.png">


9. Click on "on hold" tab, and click on Case Details
<img width="823" alt="image" src="https://user-images.githubusercontent.com/110078646/221022167-d9c1084f-60f6-4a8e-8aa0-730a4099a1de.png">

<img width="827" alt="image" src="https://user-images.githubusercontent.com/110078646/221022301-96d5999d-393c-4b10-9b8a-8044fd60c593.png">


10. From the drop down select Resend Initial Notification Letter, a modal popup will appear

<img width="830" alt="image" src="https://user-images.githubusercontent.com/110078646/221022824-efdda447-87e8-4203-8860-2d111ce1711f.png">

11. In the modal provide the instructions and click on Resend notification letter
12. The case is routed to unassigned tab and task changes to Send initial notification letter task

<img width="835" alt="image" src="https://user-images.githubusercontent.com/110078646/221023360-80ab0a50-e5f4-49ef-82d6-ae8fbb4f85ea.png">

13. Click on the case details and in the currently active task the task marks as Send Initial Notification Letter task
<img width="851" alt="image" src="https://user-images.githubusercontent.com/110078646/221023894-cf731a0d-1e3f-4fba-ba1b-d3dee977e2f9.png">

14. If you go to Case timeline, Post-Send Initial Notification Letter Holding Period task will be marked as completed
<img width="821" alt="image" src="https://user-images.githubusercontent.com/110078646/221024313-08755255-eb6a-48d7-ae1d-1ef88134f91e.png">
